src/jit/flowgraph.cpp

```diff
@@ -3883,15 +3883,38 @@ bool Compiler::fgCreateGCPoll(GCPollType pollType, BasicBlock* block)
         pollType = GCPOLL_CALL;
     }
 
+#ifdef DEBUG
+    // If a statment was supplied it should be contained in the block.
+    if (stmt != nullptr)
+    {
+        bool containsStmt = false;
+        for (Statement* stmtMaybe : block->Statements())
+        {
+            containsStmt = (stmtMaybe == stmt);
+            if (containsStmt)
+            {
+                break;
+            }
+        }
+
+        noway_assert(containsStmt);
+    }
+#endif
+
     if (GCPOLL_CALL == pollType)
     {
         createdPollBlocks = false;
         GenTreeCall* call = gtNewHelperCallNode(CORINFO_HELP_POLL_GC, TYP_VOID);
         GenTree*     temp = fgMorphCall(call);
 
-        // for BBJ_ALWAYS I don't need to insert it before the condition.  Just append it.
-        if (block->bbJumpKind == BBJ_ALWAYS)
+        if (stmt != nullptr)
+        {
+            Statement* newStmt = gtNewStmt(temp);
+            fgInsertStmtAfter(block, stmt, newStmt);
```

Comment:
the comment in the previous else case is:
`// For DDB156656, we need to associate the GC Poll with the IL offset (and therefore sequence`
do we have to restore that IL offset here as well?

Reply:
I don't think so, but it doesn't cost much so will do that.